From 1c9582b8e05aaef6f6234a15f91a31b94a9d96ec Mon Sep 17 00:00:00 2001
From: nushanth <nushanth@qti.qualcomm.com>
Date: Thu, 5 Mar 2026 23:17:38 +0530
Subject: [PATCH 6/9] Skip unused STA detach and FD re-registration in
 CONNECTED FSM

Commented out the unused station (STA HAL instances) reset and FD re-registration
logic for active STA in the CONNECTED FSM to prevents missing the EndPoint Object
change event in sta_wlan_hal_whm during BSTA disconnect.

Signed-off-by: nushanth <nushanth@qti.qualcomm.com>
---
 .../slave/backhaul_manager/backhaul_manager.cpp      | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
index 16135bacb..3382c7ae4 100644
--- a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
+++ b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
@@ -568,7 +568,9 @@ bool BackhaulManager::handle_backhaul_connect()
     // will be overriden in the loop below.
     db->backhaul.backhaul_bssid = {};
 
-    for (auto &radio_info : m_radios_info) { // Detach from unused stations first
+    //This code segment is commented out to avoid missing EndPoint Object change event, when
+    //BSTA disconnect.
+    /*for (auto &radio_info : m_radios_info) { // Detach from unused stations first
         if (db->backhaul.connection_type == AgentDB::sBackhaul::eConnectionType::Wireless &&
             ((radio_info->sta_iface == db->backhaul.selected_iface_name) &&
              (radio_info->radio_mac != net::network_utils::ZERO_MAC &&
@@ -581,7 +583,7 @@ bool BackhaulManager::handle_backhaul_connect()
                 radio_info->sta_wlan_hal.reset();
             }
         }
-    }
+    }*/
 
     for (auto &radio_info : m_radios_info) {
         if (db->backhaul.connection_type == AgentDB::sBackhaul::eConnectionType::Wireless &&
@@ -593,7 +595,9 @@ bool BackhaulManager::handle_backhaul_connect()
             db->backhaul.backhaul_bssid =
                 tlvf::mac_from_string(radio_info->sta_wlan_hal->get_bssid());
 
-            if (!radio_info->sta_wlan_hal->unique_file_descriptors() &&
+            //This code segment is commented out to avoid missing EndPoint Object change event, when
+	    //BSTA disconnect.
+	    /*if (!radio_info->sta_wlan_hal->unique_file_descriptors() &&
                 radio_info->sta_hal_ext_events.empty()) {
                 // previous iteration destroyed the sta_wlan_hal instances that are unused
                 // if the FDs were registered with another sta_wlan_hal, re-register now
@@ -661,7 +665,7 @@ bool BackhaulManager::handle_backhaul_connect()
                     return false;
                     // beerocks_agent will not receive ACTION_BACKHAUL_CONNECTED_NOTIFICATION
                 }
-            }
+            }*/
         }
     }
 
-- 
2.34.1

